### ISSUE TITLE
Curriculum Reboot

### ISSUE THREAD

**Issue by ultrasaurus** [rustbridge/reboot#3]
TODO
- [ ] try out: https://github.com/rustbridge/eurorust-2023
   - [x] check out presentations - https://github.com/rustbridge/eurorust-2023/issues
   - [ ] part 1: 
       - [ ] intro to Rust (try steps for the one in this repo
       - [ ] take a look at https://github.com/rustbridge/a-very-brief-intro-to-rust (this is what we want to merge into)
   - [ ] part 2: intro to CLIs
- [ ] come up with a plan, so other folks can help
   - [ ] All "curriculum pieces" should have a minimum time investment and a maximum time investment (so that we can say "this will take min. X hours, max. Y hours)
   - [ ] Write up next stuff, as we understand more of the plan

TBD:  (items copied from https://github.com/rustbridge/reboot/issues/1)  
- Move to a new SSG, ideally the same one that the website ends up using 
-  Snake game project: Modernize? Move to unified "docs" infrastructure maybe? Need to look at this more closely
- Condense a bit?
- See if we can't unify the presentational tool for the slides for each material

**Comment by ultrasaurus** [comment]
@shadows-withal eurorust-2023 seems to have a copy of [intro to Rust](https://github.com/ashleygwilliams/a-very-brief-intro-to-rust) -- if so, is the goal to submit them upstream to @ashleygwilliams ?   ...or create a fork of that repo here and maintain as part of the rustbridge org?

update: seems like y'all are using [remark](https://github.com/remarkjs/remark) for slides... so maybe the https://github.com/rustbridge/eurorust-2023 repo is just the output and source lives elsewhere?

**Comment by manyinsects** [comment]
@ultrasaurus Yeah, that entire repository is basically a hacked-together "oh this could work given the time constraint that we have" (we had a little bit less than 6 hours for the entire thing) kind of mini-curriculum. The original intro to rust thing is definitely outdated, and the one in the EuroRust repo is a bit more up-to-date with current practices, so I think wholesale copying that into the original repo isn't the worst approach.

(Also yes, we used remark, but we literally just wrote pure remark source files because it was faster lol)

**Comment by manyinsects** [comment]
(to be clear, by "original repo", I mean the one in this org)

**Comment by ultrasaurus** [comment]
thanks for the history @shadows-withal -- very helpful!   I've added looking at https://github.com/rustbridge/a-very-brief-intro-to-rust to the TODO checklist.  

I didn't see s repo for the CLI project -- so, assuming that curriculum will move into its own repo.... or consider some kind of unified docs structure?   (kinda partial to separate repos so students can easily clone what they are are focused on, but also can see than maintaining all the SSG content bits might become arduous, so will ponder this while I come up to speed on the specific lessons.)

**Comment by manyinsects** [comment]
obviously super delayed here, but yeah, I would like all curriculums to be in their separate repos, and I think we can find _some_ way to unify the SSG infra so that doing this shouldn't be too arduous

**Comment by ultrasaurus** [comment]
sorry to be out of touch for last few *months* -- I've gone down a few rabbit holes, which was all-well-and-good for my personal learning, but I think the project would be well-served if someone else where to take on the breakdown of tasks.

That said, happy to continue to work on this if no one else can provide better direction. So far i have:
1) Looked at Zola.  Not my fave approach for static site generators -- I much prefer a more strict separation of content and code, yet totally am in favor of whatever this community chooses, and I can learn whatever y'all decide to use
2) Looked at Intro to Rust and ran into some issues that I resolved, yet didn't take detailed notes -- mostly feel like I got a reasonable overview of what is going on

Next steps: 
1. do a detailed hands-on walk through of current repo and make note of what's broken/not working, 
2. hands-on walk through of new repo
3. merge!

FYI -- I'm pretty good at Rust for small toy projects, but don't feel like I could effectively debug a merge if I don't do step 1.  If someone thinks they could do the merge quickly and wants to give me a smaller task or ask me to QA something, happy to be redirected!
